code/game/objects/items/wayfinding.dm

```diff
@@ -1,121 +1,249 @@
+								"Voted tenth best pinpointer in the universe in 2560!", //there were no more than ten pinpointers in the game in 2020
+								"Helping assistants find the departments they tide since 2560.", //not really but it's advertising
+								"These pinpointers are flying out the airlock!", //because they're being thrown into space
+								"Grey pinpointers for the grey tide!", //I didn't pick the colour but it works
+								"Feeling lost? Find direction.",
+								"Automate your sense of direction. Buy a wayfinding pinpointer today!",
+								"Feed me a stray pinpointer.", //American Psycho reference
+								"We need a slogan!") //Liberal Crime Squad reference
+	///Number of the list entry of the slogan we're up to.
+	var/slogan_entry = 0
+	///Next tick we can say a slogan.
+	COOLDOWN_DECLARE(next_slogan_tick)
+	///Next tick the dispenser's spew rejection of non-wayfinding pinpointers can be triggered.
+	COOLDOWN_DECLARE(next_spew_tick)
 
 /obj/machinery/pinpointer_dispenser/Initialize(mapload)
 	. = ..()
-	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(ACCOUNT_CIV)
+	var/datum/bank_account/civ_acc = SSeconomy.get_dep_account(payment_department)
 	if(civ_acc)
 		synth_acc.transfer_money(civ_acc, start_bal) //float has to come from somewhere, right?
 
 	synth_acc.account_holder = name
 
-	desc += " Only [ppt_cost] credits! It also likes making costumes..."
+	desc += " [ppt_cost ? "Only [ppt_cost] credits! " : ""]It also synthesises costumes for some reason."
 
-	set_expression("neutral")
+	power_change()
+
+	COOLDOWN_START(src, next_slogan_tick, COOLDOWN_SLOGAN)
+	slogan_list = shuffle(slogan_list) //minimise repetition
+
+/obj/machinery/pinpointer_dispenser/power_change()
+	. = ..()
+	cut_overlays()
+	if(powered())
+		set_expression("veryhappy", 2 SECONDS) //v happy to be back in the pinpointer business
+		START_PROCESSING(SSmachines, src)
+
+/obj/machinery/pinpointer_dispenser/update_icon_state()
+	if(machine_stat & BROKEN)
+		set_light(0)
+	else if(powered())
+		set_light(1.4)
+	else
+		set_light(0)
+
+/obj/machinery/pinpointer_dispenser/process(delta_time)
+	if(machine_stat & (BROKEN|NOPOWER))
+		return PROCESS_KILL
+
+	if(!length(slogan_list) || !COOLDOWN_FINISHED(src, next_slogan_tick))
+		return
+	if(++slogan_entry > length(slogan_list))
+		slogan_entry = 1
+	var/slogan = slogan_list[slogan_entry]
+	say(slogan)
+	COOLDOWN_START(src, next_slogan_tick, COOLDOWN_SLOGAN)
+
+/obj/machinery/pinpointer_dispenser/Destroy()
+	for(var/i = 0, i < pick(3,9), i++) //Doesn't synthesise them in real time and instead stockpiles completed ones (though this is not how the cooldown works)
+		new /obj/item/pinpointer/wayfinding (loc)
+	say("Ouch.")
+	//An inexplicable explosion is never not funny plus it kind of explains why the machine just disappears
+	explosion(src, devastation_range = 0, heavy_impact_range = 0, light_impact_range = 1, flash_range = 3, flame_range = 1, smoke = TRUE)
```

Comment:
```suggestion
	if(!isnull(loc))
		explosion(get_truf(src), devastation_range = 0, heavy_impact_range = 0, light_impact_range = 1, flash_range = 3, flame_range = 1, smoke = TRUE)
```
Exploding the source object carries the risk of trying to delete again a deleted object.
Plus, the object could be in nullspace.